base_layer/wallet/src/transaction_service/protocols/transaction_broadcast_protocol.rs

```diff
@@ -0,0 +1,480 @@
+                self.id,
+                completed_tx.transaction.body.kernels()[0]
+                    .excess_sig
+                    .get_signature()
+                    .to_hex()
+            );
+            trace!(target: LOG_TARGET, "{}", completed_tx.transaction);
+
+            // Send Mempool Request
+            let mempool_request = MempoolProto::MempoolServiceRequest {
+                request_key: completed_tx.tx_id,
+                request: Some(MempoolProto::mempool_service_request::Request::SubmitTransaction(
+                    completed_tx.transaction.clone().into(),
+                )),
+            };
+
+            self.resources
+                .outbound_message_service
+                .send_direct(
+                    self.base_node_public_key.clone(),
+                    OutboundEncryption::None,
+                    OutboundDomainMessage::new(TariMessageType::MempoolRequest, mempool_request.clone()),
+                )
+                .await
+                .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
+
+            // Send Base Node query
+            let mut hashes = Vec::new();
+            for o in completed_tx.transaction.body.outputs() {
+                hashes.push(o.hash());
+            }
+
+            let request = BaseNodeRequestProto::FetchUtxos(BaseNodeProto::HashOutputs { outputs: hashes });
+            let service_request = BaseNodeProto::BaseNodeServiceRequest {
+                request_key: self.id,
+                request: Some(request),
+            };
+            self.resources
+                .outbound_message_service
+                .send_direct(
+                    self.base_node_public_key.clone(),
+                    OutboundEncryption::None,
+                    OutboundDomainMessage::new(TariMessageType::BaseNodeRequest, service_request),
+                )
+                .await
+                .map_err(|e| TransactionServiceProtocolError::new(self.id, TransactionServiceError::from(e)))?;
+
+            let mut delay = delay_for(self.timeout).fuse();
+            let mut received_mempool_response: Option<MempoolServiceResponse> = None;
+            let mut received_base_node_response: Option<BaseNodeProto::BaseNodeServiceResponse> = None;
+            futures::select! {
+                mempool_response = mempool_response_receiver.select_next_some() => {
+                    received_mempool_response = Some(mempool_response);
+                },
+                base_node_response = base_node_response_receiver.select_next_some() => {
+                    received_base_node_response = Some(base_node_response);
+                },
+                () = delay => {
+                },
+            }
+
+            // Handle a received Mempool Response
+            if let Some(response) = received_mempool_response {
+                if self.handle_mempool_response(response).await? {
```

Comment:
Why not have this inside the `select!` branch above?

Reply:
artifact of a previous implementation, done.